lib/backends/kv-backend.js

```diff
@@ -20,62 +20,64 @@ class KVBackend extends AbstractBackend {
    * @param {string} data[].name - Kubernetes Secret property name.
    * @param {string} data[].property - If the backend secret is an
    *   object, this is the property name of the value to use.
-   * @param {string} roleArn - If the client should assume a role before fetching the secret
+   * @param {Object} specOptions - Options set on spec level.
    * @returns {Promise} Promise object representing secret property values.
    */
-  _fetchDataValues ({ data, roleArn }) {
-    return Promise.all(data.map(async secretProperty => {
-      this._logger.info(`fetching secret property ${secretProperty.name} with role: ${roleArn || 'no role set'}`)
-      const plainOrObjValue = await this._get({ secretKey: secretProperty.key, roleArn })
-      const shouldParseValue = 'property' in secretProperty
+  _fetchDataValues ({ data, specOptions }) {
+    return Promise.all(data.map(async ({ name, property = null, key, ...keyOptions }) => {
+      const plainOrObjValue = await this._get({ key, keyOptions, specOptions })
+      const shouldParseValue = !!property
```

Comment:
total nit: I'd favor `Boolean(property)` to coerce falsy and truthy values to booleans.